### ISSUE TITLE
Hot reload Style Pointer problem

### ISSUE THREAD

**Issue by BrianBu01** [dotnet/maui#13218]
### Description

In debugging with Maui and an Android emulator, I noticed a pointer mistake with the resource dictionaries.

### Steps to Reproduce

Debugging an application (only tested with android)

Have two UI controls use the same Style definition. (I was using Frames)

Remove the style definition from one of the UI controls and the styles directly on that object.

Both object's Styles will be updated together.

### Link to public reproduction project repository

na

### Version with bug

7.0 (current)

### Last version that worked well

7.0 (current)

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android emulator

### Did you find any workaround?

Restart debugging

### Relevant log output

```shell
na
```

**Comment by drasticactions** [comment]
Can you create a reproduction project showing this? That would help me debug it.

**Comment by BrianBu01** [comment]
@drasticactions,

I wish I had time to, but currently I'm on a time crunch.  

I saw the bug, realized what was happening, and confirmed my assumption when restarting debugging fixed the problem.

I wanted to definitely report the issue, but can provide more at this time.  I would have forgotten otherwise.

Additional information that maybe relevant.

Resource Dictionary : Custom Style1

View1 -> Frame (Custom Style1) -> Grid -> CollectionView -> VertStack -> Grid -> CollectionView -> Grid -> CollectionView -> View2

View2 -> Frame (Custom Style1) -> Grid

I deleted View1.Frame.Style, then applied the style directly and it updated View2.Frame Style
